ext/memcached_rb/memcached_rb.c

```diff
@@ -1,460 +1,518 @@
-  uint32_t port = NUM2UINT(rb_port);
-
-  rb_ret = rb_str_new("", 0);
-  rb_str_append(rb_ret, rb_hostname);
-  if(0 != port) {
-    rb_str_append(rb_ret, rb_str_new2(":"));
-    rb_str_append(rb_ret, rb_fix2str(rb_port, 10));
-  }
-
-  return rb_ret;
+	UnwrapMemcached(self, mc);
+
+	value = memcached_callback_get(mc, MEMCACHED_CALLBACK_NAMESPACE, &rc);
+	rb_memcached_error_check(rc);
+
+	return value ? rb_str_new2(value) : Qnil;
 }
 
 void Init_memcached_rb(void)
 {
-  rb_mMemcached = rb_define_module("Memcached");
-  int i;
-
-  rb_eMemcachedError = rb_define_class_under(rb_mMemcached, "Error", rb_eStandardError);
-  //rb_eMemcachedErrors[0] = Qnil;
-  for(i = 1; i < MEMCACHED_ERROR_COUNT; i++){
-    const char* klass = MEMCACHED_ERROR_NAMES[i];
-    if(NULL == klass)
-      rb_eMemcachedErrors[i] = Qnil;
-    else
-      rb_eMemcachedErrors[i] = rb_define_class_under(rb_mMemcached, klass, rb_eMemcachedError);
-  }
-
-  VALUE cConnection = rb_define_class_under(rb_mMemcached, "Connection", rb_cObject);
-  rb_define_alloc_func(cConnection, allocate);
-
-  Init_memcached_rb_behavior();
-
-  rb_define_method(cConnection, "initialize", rb_connection_initialize, 1);
-  rb_define_method(cConnection, "servers", rb_connection_servers, 0);
-  rb_define_method(cConnection, "flush", rb_connection_flush, 0);
-  rb_define_method(cConnection, "set", rb_connection_set, 4);
-  rb_define_method(cConnection, "get", rb_connection_get, 1);
-  rb_define_method(cConnection, "get_multi", rb_connection_get_multi, 1);
-  rb_define_method(cConnection, "delete", rb_connection_delete, 1);
-  rb_define_method(cConnection, "add", rb_connection_add, 4);
-  rb_define_method(cConnection, "increment", rb_connection_inc, 2);
-  rb_define_method(cConnection, "decrement", rb_connection_dec, 2);
-  rb_define_method(cConnection, "exist", rb_connection_exist, 1);
-  rb_define_method(cConnection, "replace", rb_connection_replace, 4);
-  rb_define_method(cConnection, "prepend", rb_connection_prepend, 4);
-  rb_define_method(cConnection, "append", rb_connection_append, 4);
-  rb_define_method(cConnection, "get_behavior", rb_connection_get_behavior, 1);
-  rb_define_method(cConnection, "set_behavior", rb_connection_set_behavior, 2);
-  rb_define_method(cConnection, "set_prefix", rb_connection_set_prefix, 1);
-  rb_define_method(cConnection, "get_prefix", rb_connection_get_prefix, 0);
-
-  rb_cServer = rb_define_class_under(rb_mMemcached, "Server", rb_cObject);
-  rb_define_method(rb_cServer, "to_s", rb_server_to_s, 0);
-
-  rb_define_attr(rb_cServer, "hostname", 1, 0);
-  id_ivar_hostname = rb_intern("@hostname");
-  rb_define_attr(rb_cServer, "port", 1, 0);
-  id_ivar_port = rb_intern("@port");
-  rb_define_attr(rb_cServer, "weight", 1, 0);
-  id_ivar_weight = rb_intern("@weight");
-
-  id_tcp = rb_intern("tcp");
-  id_socket = rb_intern("socket");
+	size_t i;
+
+	rb_mMemcached = rb_define_module("Memcached");
+	rb_cServer = rb_const_get(rb_mMemcached, rb_intern("Server"));
```

Comment:
Don't create a `Server` class, just pick up the one we defined in Rubyland already :)